main/background.ts

```diff
@@ -87,15 +92,24 @@ ipcMain.on("metrics", async (event) => {
       }
     }
   }
-  const metrics: Metrics = {
+  const weave_size = metric_string_parse(
+    parsed.find((item: MinorParser) => item.name === "weave_size"),
+  );
+  console.log("DEBUG: getMetrics complete");
+  return {
     data_unpackaged,
     data_packaged,
+    storage_available,
+    weave_size,
     hash_rate,
     earnings,
     vdf_time_lower_bound,
   };
+}
 
-  event.reply("metrics", metrics);
+const cached_metrics: Promise<Metrics> = getMetrics();
```

Comment:
this is evaluated only once? Why not call getMetrics() in the event callback and have the renderer cache the results if/when needed?

Reply:
This is for development purposes
Later it should be reworked with pub-sub (maybe)

Reply:
ok all good :)